src/context-popup/index.tsx

```diff
@@ -0,0 +1,82 @@
+import { create, tsx } from '@dojo/framework/core/vdom';
+import focus from '@dojo/framework/core/middleware/focus';
+import { RenderResult } from '@dojo/framework/core/interfaces';
+import Popup from '../popup';
+import { createICacheMiddleware } from '@dojo/framework/core/middleware/icache';
+
+import * as css from '../theme/default/context-popup.m.css';
+
+export interface ContextPopupProperties {
+	onClose?(): void;
+	onOpen?(): void;
+}
+
+export interface ContextPopupChildren {
+	trigger: () => RenderResult;
+	content: (close: () => void, shouldFocus: () => boolean) => RenderResult;
+}
+
+interface ContextIcache {
+	x: number;
+	y: number;
+	open: boolean;
+}
+
+const icache = createICacheMiddleware<ContextIcache>();
+
+const factory = create({ icache, focus })
+	.properties<ContextPopupProperties>()
+	.children<ContextPopupChildren>();
+
+const CursorWidth = 2;
+const CursorHeight = 4;
+
+export const ContextPopup = factory(function({
+	properties,
+	children,
+	middleware: { icache, focus }
+}) {
+	const { onOpen, onClose } = properties();
+	const x = icache.getOrSet('x', 0);
+	const y = icache.getOrSet('y', 0);
+
+	const { trigger, content } = children()[0];
+	const close = () => {
+		icache.set('open', false);
+		onClose && onClose();
```

Comment:
I think the best practice is to re-read from `properties()` when you aren't directly in the render function (as that would read updated properties).  So `const { onClose } = properties();` at the top here